### PR TITLE
dependency on backbone.base-router

### PR DESCRIPTION
**Pull request by samandmoore**
goodness in the route parsing

@dschaub this was so easy it should have been illegal. and it resolves this: https://github.com/Betterment/backbone.blazer/issues/3

~~i think we should do it. it's a cheap dependency to take on given that we're delivering our stuff via bower.~~ I decided to see what it would look like to internalize the dependency. thoughts?

### CODE REVIEW

File: test/backbone.blazer.base-router.spec.js [hunk 1]
```diff
@@ -0,0 +1,640 @@
...
+
+    it('should pass an empty object for query params', function() {
+      expect(this.routeData.query).to.deep.equal({});
+    });
+
+    it('should pass empty named params', function() {
+      expect(this.routeData.params).to.deep.equal({});
+    });
+  });
+});
```
Comment (dschaub) [inline test/backbone.blazer.base-router.spec.js]: this is a base-router test right? why do we want it in our tests also?

**Comment by dschaub** [comment]
yeah, I think it makes sense, gives us a whole lot of stuff for free so why not.

File: test/backbone.blazer.base-router.spec.js (hunk 1 above)
Comment (samandmoore) [inline test/backbone.blazer.base-router.spec.js]: i wanted to make sure that we didn't break the contract that base-router provided. i think now that it's internalized as a dependency in this PR that maybe it's even more valuable to have these tests, no?

Comment (samandmoore) [inline test/backbone.blazer.base-router.spec.js]: maybe it's not necessary. it's certainly unorthodox.

**Comment by samandmoore** [comment]
@jmeas we're doing some interesting things in here. wondering if you have any thoughts.

options:
- take backbone.base-router as a bower dependency, forcing consumers of this lib to require it in their config (https://github.com/Betterment/backbone.blazer/commit/aba49e00490b4b59538d60a8c61d326d6e4d81c7)
- internalize the dependency on backbone.base-router by making it a dev dependency and merging it into our distribution file (https://github.com/Betterment/backbone.blazer/commit/e4c57a505ef052f50d4eb52259d0d61bc0d78e08)
- take no dependency, clone the functionality of backbone.base-router that we want (routeData enhancements).

**Comment by jamesplease** [comment]
I think all 3 options are good choices in different circumstances, @samandmoore. Generally I use the 3rd option if there's a single algorithm that I want to pluck from a library that otherwise does nothing I need. I haven't taken a close-enough look at this PR to see if that applies here, but it very well might!

As for deciding between 1 and 2, I've been thinking of that very question myself. I'd probably go with the first myself, since it's easiest for me and developers using npm-based module bundlers (the group I cater to).

File: test/backbone.blazer.base-router.spec.js (hunk 1 above)
Comment (jamesplease) [inline test/backbone.blazer.base-router.spec.js]: Mmm...I don't think that these are necessary. Also, the maintenance would be a nightmare.

**Comment by samandmoore** [comment]
gonna meditate on this for a bit. i think i'm just going to implement the bits that we want internal to blazer. thanks for the input guiseee.

**Pull request closed without merging**